ssr-express/README.md

```diff
@@ -0,0 +1,131 @@
+<!---
+Copyright 2018 The AMP HTML Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS-IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+-->
+
+## Introduction
+
+amp-ssr-express is an [express](http://expressjs.com/) middleware that applies AMP server-side-rendering on the fly, using the [amp-ssr](../ssr) component.
+
+For the middleware to work, a route that renders AMP pages must be available. It will then rewrite the
+request to that route and apply server-side-rendering to the result.
+
+As an example, a website that has an SSR page at `https://example.com/page.html` would have it's AMP
+equivalent at the route `https://example.com/amp/page.html`. The middleware will intercept the original request to `/page.html` rewrite it to `/amp/page.html` and transform the output.
+
+Requests made directly to the AMP page, such as `https://example.com/amp/page.html` will not have
+server-side-rendering applied.
+
+## Usage
+
+Usage of this middleware follows the standard for most Express middlewares. It is important that the
+middleware is used *before* the middleware or route that renders the page.
+
+The example bellow will transform HTML being loaded by express-static:
+```javascript
+const ampSsrMiddleware = createAmpSsrMiddleware(ampSsr);
+const staticMiddleware = express.static(path.join(__dirname, '/public'));
+
+app.use(ampSsrMiddleware);
+app.use(staticMiddleware);
+```
+
+If `staticMiddleware` was used before `transformMiddleware`, the request would be finished after the first
+finishes, and the page wouldn't have a change to be transformed.
+
+### Using the default configuration
+
+The default configuration uses the `/amp/` as the prefix for AMP pages. The two effects from this are:
+- Pages which the pathname start with `/amp/` will not have the transformation applied.
+eg: `https://example.com/amp/page.html`.
+- Pages that do not start with `/amp/` will have the URL rewritten to the equivalent page by appending
+the prefix to URL and the output will be transformed. eg: `https://example.com/page.html` is rewritten
+to `https://example.com/amp/page.html`.
+
+Example configuration using the default configuration and a static middleware:
+```javascript
+const express = require('express');
+const path = require('path');
+const app = express();
+const createAmpSsrMiddleware = require('amp-toolbox-ssr-express');
+const ampSsr = require('amp-toolbox-ssr');
+
+ampSsr.setConfig({
```

Comment:
Is this config global, or local to each middleware instance? (If you want two for some reason.)

Reply:
amp-toolbox-ssr exports the following:
`module.exports = new DomTransformer(defaultConfig);`

So, every time it is required, a new DomTransformer object is created.

Reply:
Are you sure about that? `require(.)` is cached: "[every call to require('foo') will get exactly the same object returned](https://nodejs.org/docs/latest/api/modules.html#modules_caching))."

Reply:
Hum. Didn't know that. We may want to change `amp-ssr` if we want to have multiple instances then.

Reply:
+1